handler.go

```diff
@@ -233,6 +240,45 @@ func mountStaticFiles(logger *slog.Logger, mux *http.ServeMux) error {
 	})
 }
 
+// Requires basic auth username/password authentication on all endpoints
+// https://en.wikipedia.org/wiki/Basic_access_authentication
+type basicAuthMiddleware struct {
+	username string
+	password string
+}
+
+func (m basicAuthMiddleware) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		basicAuthHeader := r.Header.Get("Authorization")
+		if basicAuthHeader == "" {
+			w.Header().Set("WWW-Authenticate", `Basic realm="riverui restricted"`)
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		if !strings.HasPrefix(basicAuthHeader, "Basic ") {
+			w.Header().Set("WWW-Authenticate", `Basic realm="riverui restricted"`)
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		data, err := base64.StdEncoding.DecodeString(basicAuthHeader[len("Basic "):])
+		if err != nil {
+			w.Header().Set("WWW-Authenticate", `Basic realm="riverui restricted"`)
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		parts := strings.Split(string(data), ":")
+		if len(parts) != 2 || parts[0] != m.username || parts[1] != m.password {
+			w.Header().Set("WWW-Authenticate", `Basic realm="riverui restricted"`)
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+		next.ServeHTTP(w, r)
+	})
+}
```

Comment:
Do you think you could take a stab at a basic couple of tests for this middleware? I know the project is hit or miss testing wise, but we're trying to shore that up.